firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentSnapshot.java

```diff
@@ -589,8 +601,25 @@ private Object convertValue(FieldValue value, FieldValueOptions options) {
             database.getDatabaseId());
       }
       return new DocumentReference(key, firestore);
+    } else if (value instanceof TimestampValue) {
+      Timestamp timestamp = ((TimestampValue) value).value();
```

Comment:
Rather than this one giant method, it would be nice to factor these out into little, easy to understand private helpers (as was done in iOS).

Reply:
Helpers added.